scope_ascom.cpp

```diff
@@ -136,7 +136,7 @@ wxArrayString ScopeASCOM::EnumAscomScopes()
     return list;
 }
 
-bool ScopeASCOM::Create(DispatchObj& obj)
+bool ScopeASCOM::Create(DispatchObj& obj, bool *alreadyRegistered)

```

Comment:
I think this will be a little easier to understand if we change the name and invert the sense of this parameter
```suggestion
// *registered is set to true if the Dispatch object was registered in the
// Global Interface Table
bool ScopeASCOM::Create(DispatchObj& obj, bool *registered)
```
so that if `registered` is returned as `true` then the caller may need to call `Unregister`

I would also suggest setting `*registrered = false` at the top of the function, then set it to `true` right after the call to `m_gitEntry.Register(obj);` as that will ensure that it never gets set to tru prematurely regardless of thrown exceptions etc.